vitals/Main.java

```diff
@@ -1,23 +1,82 @@
 package vitals;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class Main {
-    static boolean batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
-            System.out.println("Temperature is out of range!");
-            return false;
-        } else if(soc < 20 || soc > 80) {
-            System.out.println("State of Charge is out of range!");
-            return false;
-        } else if(chargeRate > 0.8) {
-            System.out.println("Charge Rate is out of range!");
-            return false;
-        }
-        return true;
+
+  /**
+   *
+   */
+  private static final String CHARGE_RATE = "Charge Rate";
+  /**
+   *
+   */
+  private static final String STATE_OF_CHARGE = "State of Charge";
+  /**
+   *
+   */
+  private static final String TEMPERATURE = "Temperature";
+
+  static Map<String, Boolean> checkVitalsStatus(final Map<String, Float> vitalValuesMap) {
+    Map<String, Boolean> vitalStatusMap = new HashMap<>();
+    vitalStatusMap.put(TEMPERATURE, checkTemperatureRange(vitalValuesMap.get(TEMPERATURE)));
+    vitalStatusMap.put(STATE_OF_CHARGE, checkChargeStatus(vitalValuesMap.get(STATE_OF_CHARGE)));
+    vitalStatusMap.put(CHARGE_RATE, checkRate(vitalValuesMap.get(CHARGE_RATE)));
+    return vitalStatusMap;
+  }
+
+  static boolean checkTemperatureRange(final float temperature) {
+    if ((temperature < 0) || (temperature > 45)) {
```

Comment:
Similar code ( in soc check) can be eliminated and hence code duplication can be reduced.